lib/building.rb

```diff
@@ -0,0 +1,51 @@
+class Building
+  attr_reader :units,
+              :renters
+
+  def initialize
+    @units = []
+    @renters = []
+  end
+
+  def add_unit(unit)
+    @units << unit
+  end
+
+  def renters
+    @units.map do |unit|
+      if unit.renter != nil
+        @renters << unit.renter.name
```

Comment:
Should have shovelled into a local variable. Youd get doubles if you ran this method twice